File: CSES/Sorting and Searching/1629 - Movie Festival/movieFestival.py
n = int(input())
end = 0
sol = 0
movies = []
 
for i in range (n):
  a, b = input().split() 
  movies.append([int(a), int(b)])
 
# ordena com base no final (2o elemento)
movies.sort(key=lambda x: x[1])
 
for movie in movies:
    if movie[0] >= end:
        end = movie[1]
        sol += 1    
 
print(sol)

File: CSES/Sorting and Searching/1629 - Movie Festival/movieFestival.cpp
#include <bits/stdc++.h>
using namespace std;

#define FOR(i,a,b) for (int i = a; i < b; i++)
#define pb push_back
#define mp make_pair
#define F first
#define S second
#define sortv(v) sort((v).begin(), (v).end())

int main() {
vector<pair<int, int>> movies;
int n, sol = 0, end = 0;
cin >> n;

FOR(i,0,n){
  int a,b; cin >> a >> b;
  movies.pb(mp(b,a));
}

sortv(movies);

for (auto movie : movies){
  if( movie.S >= end) {
    end =  movie.F;
    sol++;
  }
}

cout << sol << endl;
}
